FsSearchableSnapshotsIT trips assertion

**Build scan**: https://gradle-enterprise.elastic.co/s/yeui6gg3ga7fe

```
» [2021-03-30T16:38:46,793][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [integTest-0] fatal error in thread [elasticsearch[integTest-0][searchable_snapshots_cache_fetch_async][T#5]], exiting
»  java.lang.AssertionError: expected a non-closed channel reference
»  	at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFile$2.doRun(CacheFile.java:368) ~[?:?]
»  	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:728) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
»  	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
»  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
»  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
»  	at java.lang.Thread.run(Thread.java:832) [?:?]
```

--- a/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFile.java
+++ b/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/common/CacheFile.java
@@ -365,7 +365,6 @@ public class CacheFile {
                     @Override
                     protected void doRun() throws Exception {
                         if (reference.tryIncRef() == false) {
-                            assert false : "expected a non-closed channel reference";
                             throw new AlreadyClosedException("Cache file channel has been released and closed");
                         }
                         try {

--- a/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java
+++ b/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java
@@ -667,7 +667,6 @@ public class FrozenCacheService implements Releasable {
                         @Override
                         protected void doRun() throws Exception {
                             if (CacheFileRegion.this.tryIncRef() == false) {
-                                // assert false : "expected a non-closed channel reference";
                                 throw new AlreadyClosedException("Cache file channel has been released and closed");
                             }
                             try {

